### PR TITLE
[VL] Link lib gluten to arrow's static libraries

### PR DESCRIPTION
**Pull request by PHILO-HE**
1. Removes linking libvelox.so with arrow libs, which is useless.
2. With static arrow libs linked, no need to keep lib arrow & lib parquet in Gluten Jar.

### CODE REVIEW

**Comment by FelixYBW** [comment]
Perf regression observed

**Comment by PHILO-HE** [comment]
> Perf regression observed

@FelixYBW, seems the regression is caused by other code or some other test factors. After code rebase, TPCH's perf. is 1067.17s now.

**Review by zhouyuan — APPROVED** [review]

File: dev/build_arrow.sh [hunk 1]
```diff
@@ -38,15 +36,14 @@ function install_arrow_deps {
...
-   BUILD_TYPE=$1
- fi
  pushd $ARROW_PREFIX/cpp
 
  cmake_install \
        -DARROW_PARQUET=ON \
        -DARROW_FILESYSTEM=ON \
        -DARROW_PROTOBUF_USE_SHARED=OFF \
+       -DARROW_DEPENDENCY_USE_SHARED=OFF \
+       -DARROW_DEPENDENCY_SOURCE=BUNDLED \
```
Comment (Yohahaha) [inline dev/build_arrow.sh:46]: > -DARROW_DEPENDENCY_SOURCE=BUNDLED

does this change really necessary? some deps may already installed.

Comment (PHILO-HE) [reply dev/build_arrow.sh:46]: @Yohahaha, thanks for your comment. If not bundled, we need some extra code to support resolve those installed dependencies and link them. Considering arrow libs are generally built once then installed, currently we can simply use BUNDLED to get a lib of all arrow's bundled dependencies.

Comment (Yohahaha) [reply dev/build_arrow.sh:46]: > Considering arrow libs are generally built once then installed

it's important, hope we can reduce changes of arrow...

Comment (PHILO-HE) [reply dev/build_arrow.sh:46]: @Yohahaha, I will take a look.

**Comment by zhztheplayer** [comment]
Observed some issues in dynamic build after this change

```
[ 69%] Linking CXX executable BenchmarkCompression
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4_compress_default'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_initCStream'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_freeCStream'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_compressBound'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4_compress_HC'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_compressFrame'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_compressFrameBound'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_flush'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `inflateInit2_'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_minCLevel'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_createDecompressionContext'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_freeCompressionContext'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_decompress'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_getErrorName'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_maxCLevel'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_createCStream'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_compressBegin'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_isError'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4_decompress_safe'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_getErrorName'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `inflate'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_isError'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_resetDecompressionContext'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_createCompressionContext'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_endStream'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4_compressBound'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `LZ4F_compressionLevel_max'
/usr/bin/ld: ../../releases/libgluten.so: undefined reference to `ZSTD_compressBound'
```

The build command used is 

```
dev/builddeps-veloxbe.sh --build_type=RelWithDebInfo --build_tests=ON --build_benchmarks=ON
```

**Comment by zhztheplayer** [comment]
@PHILO-HE Update, if I change to ` --build_tests=OFF --build_benchmarks=OFF` then compilation could succeed.

**Comment by PHILO-HE** [comment]
> @PHILO-HE Update, if I change to ` --build_tests=OFF --build_benchmarks=OFF` then compilation could succeed.

@zhztheplayer, thanks for your feedback! I'll fix it.

**Comment by FelixYBW** [comment]
@PHILO-HE can you confirm if BUILD_TYPE=relWithDebInfo is passed to Arrow when you build arrow? Looks the main branch doesn't pass it.

**Comment by PHILO-HE** [comment]
> @PHILO-HE can you confirm if BUILD_TYPE=relWithDebInfo is passed to Arrow when you build arrow? Looks the main branch doesn't pass it.

@FelixYBW, currently, we always build arrow with Release type, assuming generally no need to align the build type with main project. Hongze has helped fixed some issues related to thrift. And now both arrow/velox can use system installed arrow libs. If you have any issue, please let me know.

**Pull request merged**